src/core_atmosphere/Makefile

```diff
@@ -56,10 +57,12 @@ atmcore: $(PHYSCORE) dycore diagcore $(OBJS)
 
 mpas_atm_core_interface.o: mpas_atm_core.o
 
-mpas_atm_core.o: dycore diagcore mpas_atm_threading.o
+mpas_atm_core.o: dycore diagcore mpas_atm_threading.o mpas_atm_halos.o
 
 mpas_atm_dimensions.o:
 
+mpas_atm_halos.o:
```

Comment:
Recognizing that we've set a bad example in the existing Makefiles, I think we can just omit `mpas_atm_halos.o` since it has no dependencies in the `core_atmosphere` directory or below.